internal/elasticsearch/ingest/datastream.go

```diff
@@ -98,17 +98,18 @@ func loadIngestPipelineFiles(dataStreamPath string, nonce int64) ([]Pipeline, er
 			return nil, err
 		}
 
-		c, err = addRerouteProcessors(c, dataStreamPath, path)
+		cWithRerouteProcessors, err := addRerouteProcessors(c, dataStreamPath, path)
 		if err != nil {
 			return nil, err
 		}
 
 		name := filepath.Base(path)
 		pipelines = append(pipelines, Pipeline{
-			Path:    path,
-			Name:    getPipelineNameWithNonce(name[:strings.Index(name, ".")], nonce),
-			Format:  filepath.Ext(path)[1:],
-			Content: c,
+			Path:            path,
+			Name:            getPipelineNameWithNonce(name[:strings.Index(name, ".")], nonce),
+			Format:          filepath.Ext(path)[1:],
+			Content:         cWithRerouteProcessors,
+			ContentOriginal: c,
```

Comment:
Added a new field `ContentOriginal` in this struct to be able to keep the original contents, the ones read from the ingest pipeline file defined in the package.

It is being kept `Content` to hold the same contents (containing reroute processors) to not change the current behaviour.

Reply:
Another consequence of adding those reroute processors, is that the YAML written afterwards is not the same as in the package:
- comments are not kept
- fields in the YAML do not keep the same order